lib/service/InitService.ts

```diff
@@ -0,0 +1,74 @@
+import errors from '../service/errors';
+import { EventEmitter } from 'events';
+import SwapClientManager from '../swaps/SwapClientManager';
+import NodeKey from '../nodekey/NodeKey';
+import assert = require('assert');
+
+interface InitService {
+  once(event: 'nodekey', listener: (nodeKey: NodeKey) => void): this;
+  emit(event: 'nodekey', nodeKey: NodeKey): boolean;
+}
+
+class InitService extends EventEmitter {
+  /** Whether there is a pending `CreateNode` or `UnlockNode` call. */
+  public pendingCall = false;
+
+  constructor(private swapClientManager: SwapClientManager, private nodeKeyPath: string, private nodeKeyExists: boolean) {
+    super();
+  }
+
+  public createNode = async (args: { password: string }) => {
+    const { password } = args;
+    if (this.nodeKeyExists) {
+      throw errors.UNIMPLEMENTED;
+    }
+    if (this.pendingCall) {
+      throw errors.PENDING_CALL_CONFLICT;
+    }
+
+    this.pendingCall = true;
+    const seed = await this.swapClientManager.genSeed();
+    let nodeKey: NodeKey;
+
+    if (seed) {
+      const seedBytes = typeof seed.encipheredSeed === 'string' ?
+        Buffer.from(seed.encipheredSeed, 'base64') :
+        Buffer.from(seed.encipheredSeed);
+      assert.equal(seedBytes.length, 33);
+
+      // the seed is 33 bytes, the first byte of which is the version
+      // so we use the remaining 32 bytes to generate our private key
+      const privKey = Buffer.from(seedBytes.slice(1));
```

Comment:
Yeah that might make sense. This might be a little bit hacky, and we're not getting a full 32 bytes of entropy this way, but it so happens that the aezeed key is 33 bytes of which the 1st is a near-static version byte, and the remaining 32 bytes are enough for our priv key. I'll add a TODO here.